docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+# Overview
+VM expose a different, and partially overlapping, set of metrics with
+respect to containers. This document explain the differences and how
+those metrics are made available to the cluster, to be consumed using
+the standard kubernetes facilities.
+
+## libvirt metrics
+[libvirt](https://libvirt.org/) is the virtualization framework that
+kubevirt uses; in order to describe the VM metrics in kubevirt, the
+first step is to inspect what libvirt provides, and how it collect the
+data.
+
+In a nutshell, libvirt uses both the QEMU configuration and the standard
+facilites of the linux kernel to do the resource limitation and
+accounting. As prime example, libvirt uses cgroups, to do resource
+accounting for memory, cpu usage, I/O. Thus, we can consider this subset
+of metrics to fully overlap with what monitoring system (e.g.
+[cAdvisor](https://github.com/google/cadvisor)) already provides.
+Let's call this set `system metrics`.
+
+Libvirt exposes a set of hypervisor and operation-specific metrics.
+Examples are the migration statistics. Those metrics are usually
+available only while the operation is in progress, or shortly after.
+Let's call this set `hypervisor metrics`
+
+Lastly, VMs often run a Guest Agent, a deamon which periodically
+collects and reports back to the hypervisor another set of metrics.
+Let's call this set `guest metrics`
+
+The last two sets have no counterpart in the containers ecosystem, but
```

Comment:
Both could actually be seen as application metrics in the Kube world, but yes, on the Kube level there is no coutnerpart.